include/boost/graph/rank_spanning_branchings.hpp

```diff
@@ -0,0 +1,1159 @@
+	    }
+                 
+	    BOOST_FOREACH( Vertex v, cycle_vertex_set )
+	    {
+              if( v != new_repr ) in_edges[new_repr].merge( in_edges[v] );
+            }
+
+	    unvisited_vertex_set.insert( new_repr );
+
+            parent[new_repr] = v_new;
+
+	  }
+
+        }
+
+      }
+
+      // Create a set containing possible roots of the cycle branching.
+      // In each cycle expansion, remove isolated vertices of C to avoid
+      // considering them in subsequent cycle expansions.
+
+      unordered_set<BranchingVertex> root_set;
+
+      BGL_FORALL_VERTICES( u, C, BranchingGraph )
+      {
+        root_set.insert( u );
+      }
+
+      while( !root_set.empty() )
+      {
+        expand( g, v_id, C, root_set, beta );
+      }
+
+      BGL_FORALL_VERTICES_T( v, g, Graph )
+      {
+        if( v != root_vertex )
+        {
+          branching.insert( beta[v_id[v]].edge );
+        }
+      }
+
+    }
+
+    // Depth-first visitor to set up pre-order and post-order maps.
+
+    template<typename OrderMap>
+    class dfs_order_visitor:public default_dfs_visitor
+    {
+
+      public:
+	dfs_order_visitor(
+	  OrderMap& pr,
+	  OrderMap& po
+	) :  m_pr( pr ), m_po( po ) { td = 0; tf = 0; }
+
+	template<typename Vertex, typename Graph>
+	void discover_vertex(const Vertex u, const Graph & g)
+	{
+	  m_pr[u] = td++;
+	}
+
+	template <typename Vertex, typename Graph>
+	void finish_vertex(const Vertex u, const Graph & g)
+	{
+	  m_po[u] = tf++;
+	}
+
+	OrderMap& m_pr;
+	OrderMap& m_po;
+
+      private:
+	std::size_t td;
+	std::size_t tf;
+
+    };
+
+    // The ancestor checker.  A vertex u is a proper ancestor of
+    // vertex v (in the parent branching) if pr[u] < pr[v] and po[u] > po[v].
+
+    template<typename OrderMap>
+    struct ancestor_checker
+    {
+      OrderMap& m_pr;
+      OrderMap& m_po;
+      ancestor_checker( OrderMap& pr, OrderMap& po ) :
+	m_pr( pr ), m_po( po ) {}
+
+      template<typename Vertex>
+      bool operator()( const Vertex& v1, const Vertex& v2 ) const
+      { return ( m_pr[v1] < m_pr[v2] && m_po[v1] > m_po[v2] ); }
+    };
+
+    // Camerini et al. NEXT routine.
+
+    template <typename Graph, typename Edge, typename IndexMap,
+              typename WeightMap, typename Rank, typename Pred,
+              typename Compare>
+    shared_ptr<
+      std::pair<Edge, typename property_traits<WeightMap>::value_type>
+    >
+    next_spanning_branching( const Graph& g, 
+                             const unordered_set<Edge>& branching,
+                             IndexMap& v_id,
+                             WeightMap& w,
+                             Compare& comp,
+                             Rank rank,
+                             Pred pred1,
+                             Pred pred2,
+                             unordered_set<Edge>& include_edges,
+                             unordered_set<Edge>& exclude_edges
+                           )
+    {
```

Comment:
This function, `next_spanning_branching`, is very long. I recommend you break it up along the lines of major sequences, loops and/or conditions.